firestore.rules

```diff
@@ -301,6 +301,15 @@ service cloud.firestore {
           return get(/databases/$(database)/documents/authed/$(portal)/documents/$(docId)).data.teachers;
         }
 
+        // return owner of the parent document
+        function getHistoryParentDocumentOwner() {
+         return get(/databases/$(database)/documents/authed/$(portal)/documents/$(docId)).data.uid;
+        }
+         // return context of the parent document
+        function getHistoryDocumentContext() {
+         return get(/databases/$(database)/documents/authed/$(portal)/documents/$(docId)).data.context_id;
```

Comment:
Did you look around at the use of `context_id` vs `contextId`.  I think we'd prefer `contextId`, but if that means a migration of existing data it is fine to use `context_id` for now.

Reply:
I skipped doing this since we decided to use the existing teacher access stuff instead.